src/HookListener/Registration.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+namespace Terminal42\AutoRegistrationBundle\HookListener;
+
+use Contao\CoreBundle\Monolog\ContaoContext;
+use Contao\MemberModel;
+use Doctrine\DBAL\Connection;
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Component\Security\Core\Authentication\Token\UsernamePasswordToken;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+use Symfony\Component\Security\Http\Event\InteractiveLoginEvent;
+
+
+class Registration
+{
+    /**
+     * @var UserProviderInterface
+     */
+    private $userProvider;
+
+    /**
+     * @var TokenStorageInterface
+     */
+    private $tokenStorage;
+
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var EventDispatcherInterface
+     */
+    private $eventDispatcher;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * Registration constructor.
+     *
+     * @param UserProviderInterface    $userProvider    The user provider.
+     * @param TokenStorageInterface    $tokenStorage    The token storage.
+     * @param Connection               $connection      The database connection.
+     * @param LoggerInterface          $logger          The logger.
+     * @param EventDispatcherInterface $eventDispatcher The event dispatcher.
+     * @param RequestStack             $requestStack    The request stack.
+     */
+    public function __construct(UserProviderInterface $userProvider, TokenStorageInterface $tokenStorage, Connection $connection, LoggerInterface $logger, EventDispatcherInterface $eventDispatcher, RequestStack $requestStack) {
+        $this->userProvider    = $userProvider;
+        $this->tokenStorage    = $tokenStorage;
+        $this->connection      = $connection;
+        $this->logger          = $logger;
+        $this->eventDispatcher = $eventDispatcher;
+        $this->requestStack    = $requestStack;
+    }
+
+    /**
+     * Within the registration process, log in the user if needed.
+     *
+     * @param int   $userId The user id
+     * @param array $data   The user data of the registration module
+     */
+    public function onCreateNewUser(int $userId, array $data): void
+    {
+        global $objPage;
+
+        $statement = $this->connection->createQueryBuilder()
```

Comment:
Do we need to query this from the database? Is it not already available in the `$objPage` object? Or should we rather use Models (e.g. `PageModel::findById($objPage->rootId)`?

Reply:
>  Is it not already available in the $objPage object?

No, `$objPage->auto_login_activation` would result in `""`. The value of the root page does not get inherited.

> Or should we rather use Models (e.g. PageModel::findById($objPage->rootId)

I'm pretty sure that a plain database query is faster, isn't they?

Next question, will there be a way to overcome `global $objPage` in Contao 4?

Reply:
> I'm pretty sure that a plain database query is faster, isn't they?

Not really. First of all, `PageModel::findById()` would be one DB query as well. Secondly, the object most likely/surely already is in the cache because it was needed to build the page object.

> Next question, will there be a way to overcome global $objPage in Contao 4?

Not yet / not until we implement `symfony/cmf-routing`